### PR TITLE
Optimize resolveEmailIDs for single-result commands

### LINKED ISSUE #80: Optimize resolveEmailIDs for single-result commands
When the `unsubscribe` command (and similar single-email commands) use filter flags, `resolveEmailIDs` calls `QueryEmailIDs` which paginates the full result set. Since only the first match is used (`ids[0]`), this can be unnecessarily expensive for broad filters.

Consider adding a dedicated helper (e.g., `QueryFirstEmailID` or `ResolveFirstEmailID`) that uses `Email/query` with `Limit: 1` when no explicit email ID argument is provided.

Flagged by Copilot review on PR #79.

### PR DESCRIPTION
**Pull request by cboone**
## Summary

- Add `QueryFirstEmailID` client method that issues `Email/query` with `Limit: 1` and `receivedAt` descending sort, avoiding full pagination when only the first match is needed
- Add `resolveFirstEmailID` command helper mirroring `resolveEmailIDs` for single-email commands
- Add 6 tests covering happy path, empty results, error propagation, limit/sort validation, filter passthrough, and args short-circuit

## Test plan

- [ ] `go build ./...` compiles without errors
- [ ] `go test ./...` passes all existing and new tests
- [ ] `go vet ./...` and `golangci-lint run ./...` report no issues

## Closes

Closes #80

### CODE REVIEW

File: cmd/filters.go [hunk 1]
```diff
@@ -174,6 +174,30 @@ func resolveEmailIDs(cmd *cobra.Command, args []string, c *client.Client) ([]str
 	return ids, nil
 }
 
+// resolveFirstEmailID returns a single email ID from args or queries the most
+// recent match using filter flags.
+func resolveFirstEmailID(cmd *cobra.Command, args []string, c *client.Client) (string, error) {
+	if len(args) > 0 {
```
[the remark this answers is not in the document]
Comment (cboone) [reply cmd/filters.go]: Good catch. Fixed in 61832f7: `resolveFirstEmailID` now returns an error when `len(args) > 1` instead of silently dropping extra arguments.

**Comment by cboone** [comment]
## Copilot Feedback Resolved

Addressed 1 Copilot review comment with code changes:

| File | Category | Action |
|------|----------|--------|
| `cmd/filters.go:180` | Valid | Added error for multiple email IDs instead of silently dropping extras |
